components/suggestion/command_provider/applet_command_parser.ts

```diff
@@ -0,0 +1,589 @@
+        const b = {
+            ...binding,
+            func: res.data,
+        }
+        COMMANDS.push(b);
+        return b;
+    }
+}
+
+export const getCloudAppCommandLocations = (state: GlobalState): AppletBinding[] => {
+    // return STATIC_BINDINGS;
+    const bindings = getPluginsLocations(state, "/command");
+    return bindings;
+}
+
+export const getCloudAppCommand = (name: string): AppletBinding | void => {
+    return COMMANDS.find((c) => c.name === name);
+}
+
+export const getCloudAppCommands = (): AppletBinding[] => {
+    return COMMANDS;
+}
+
+export const handleCommand = async (pretext: string, bindings: AppletBinding[], args: {channel_id: string; root_id?: string;}): Promise<SuggestionChoice[]> => {
+    ARGS = {
+        app_id: '',
+        acting_user_id: '',
+        channel_id: args.channel_id,
+        root_post_id: args.root_id,
+    };
+
+    const base = extractBaseCommand(pretext);
+    let binding = getCloudAppCommand(base);
+    if (!binding) {
+        binding = bindings.find((b) => b.name === base);
+        if (!binding) {
+            return [];
+        }
+        if (binding.call) {
+            binding = await fetchCloudAppCommand(binding);
+        }
+    }
+
+    if (!binding) {
+        return [{suggestion: 'oh no'}]
+    }
+    const res = await getCurrentlyEditingToken(pretext, [binding]);
+    return res.map((sug) => decorateSuggestionChoiceComplete(pretext, sug));
+}
+
+export const getSynchronousResults = (pretext: string, bindings: AppletBinding[]): SuggestionChoice[] => {
+    const result: SuggestionChoice[] = [];
+    for (const binding of bindings) {
+        let trigger = binding.name;
+        if (trigger[0] !== '/') {
+            trigger = '/' + trigger;
+        }
+        if (trigger.startsWith(pretext)) {
+            result.push(decorateSuggestionChoiceComplete(pretext, {
+                complete: trigger,
+                suggestion: trigger,
+                // description: cmd.description,
+                // hint: cmd.pretext,
+            }));
+        }
+    }
+
+    return result;
+}
+
+export const getCurrentlyEditingToken = async (pretext: string, bindings: AppletBinding[]): Promise<SuggestionChoice[]> => {
+    if (!pretext.length) {
+    }
+
+    let cmd = getMatchedCommand(pretext, bindings);
+
+    if (!cmd) {
+        return [];
+    }
+
+    let suggestions: SuggestionChoice[] = [];
+    if (cmd.bindings && cmd.bindings.length) {
+        const subSuggestions = getSuggestionsForSubCommands(pretext, cmd);
+        suggestions = suggestions.concat(subSuggestions);
+    } else {
+        // Add "Execute Current Command" suggestion
+        const execute = getSuggestionForExecute(pretext);
+        suggestions = [execute, ...suggestions];
+    }
+
+    if (cmd.call && !cmd.func) {
+        cmd = await fetchCloudAppCommand(cmd);
+    }
+
+    if (cmd.func && cmd.func.form) {
+        const argSuggestions = await getSuggestionsForArguments(pretext, cmd);
```

Comment:
Seems like the `Execute` logic should go here? Yet, somehow it's a special item at the top of the list, when available...